### DISCUSSION TITLE
dedup across different datasets with different recordsizes

### DISCUSSION THREAD

**Discussion by christianmerges** [openzfs/zfs#15286] [Q&A] [answered]
Hello,

as far as i understood. Dedup is happening at pool level. When i create a dataset with a different recordsize. I assume that this data is not deduped against the other files in the pool, because the dedup table is created by generating a checksum of each block?

Just wanted to double check on that.

**Comment by amotin** [comment, marked as answer]
Right. Dedup may still happen for files smaller than the smallest recordsize (and so having only one block), but otherwise each dataset with different recordsize will be on its own.
